ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/compare/ManifestStructureCreator.java

```diff
@@ -222,14 +221,7 @@ public boolean isEditable() {
 				return isEditable;
 			}
 		};
-		try {
-			parseManifest(rootNode, document, monitor);
-		} catch (IOException ex) {
-			if (adapter != null)
-				adapter.disconnect(input);
-			throw new CoreException(Status.error(PDEUIMessages.ManifestStructureCreator_errorMessage, ex));
-		}
-
+		parseManifest(rootNode, document, monitor);
```

Comment:
Does the disconnection happen somewhere else?

Reply:
The IOException here was never thrown so the adapter already was never disconnected. if such disconnect is wanted please catch RuntimeException